engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/PropertyMetaData.java

```diff
@@ -262,7 +262,14 @@ else if ( constrainedElement.getKind() == ConstrainedElementKind.METHOD ) {
 
 			// fast track if it's a regular constraint
 			if ( !(constraint.getLocation() instanceof TypeArgumentConstraintLocation) ) {
-				converted = getterConstraintLocation;
+				// If the declaring class is a sub-class that has inherited the getter, ensure the location is preserved during conversion.
+				// RD: This is to address HV-1534
+				if ( requiresConversionToGetterLocation( getterConstraintLocation, constraint ) ) {
+					converted = getterConstraintLocation;
+				}
+				else {
+					converted = constraint.getLocation();
+				}
```

Comment:
OK, so this is an ugly hack so if it ends up to it, it's a no-no for me.

This method is used to convert other type of constraint location to a getter constraint location so if it is already one, we do not need to do anything.

So I would simply do:
```
if ( constraint.getLocation() instanceof GetterConstraintLocation ) {
    converted = constraint.getLocation();
}
else {
    converted = getterConstraintLocation;
}
```

and the same for the other spot you forgot at line 293 of your updated code.

Reply:
Hi @gsmet thank you kindly for reviewing this PR so quickly. Appreciate this deviates from your original discussions. I will attempt to make changes and update the PR as soon as I'm able (hopefully tonight).

My usual preference is to move logic (along these lines) into a private static method in the class to improve readability, but I'm obviously happy to follow your direction on this.

I tried a couple of different checks to determine if the ConstraintLocation needed to be converted to the new GetterConstraintLocation and they caused other test failures. I will try again with your suggestion and if there are any failures (where fixes aren't apparent) I'll update in a comment here. Thanks.

Reply:
> My usual preference is to move logic (along these lines) into a private static method in the class to improve readability, but I'm obviously happy to follow your direction on this.

Agreed when it hides some complexity and allows to understand the logic better. If we end up with my proposal, I think it's better not to hide anything considering it's very simple and natural.

> I tried a couple of different checks to determine if the ConstraintLocation needed to be converted to the new GetterConstraintLocation and they caused other test failures. I will try again with your suggestion and if there are any failures (where fixes aren't apparent) I'll update in a comment here.

OK, let's try with my proposal. If it doesn't work, I'll have a look. But if we end up with complicated things, either we find a better solution or it will be a bit too hackyish for me I'm afraid.

Reply:
Happy to proceed as you suggest.